extensions/typescript/src/features/jsDocCompletionProvider.ts

```diff
@@ -0,0 +1,184 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import { Position, Selection, Range, CompletionItemProvider, CompletionItemKind, TextDocument, CancellationToken, CompletionItem, window, commands, Uri, ProviderResult, TextEditor } from 'vscode';
+
+import { ITypescriptServiceClient } from '../typescriptService';
+import { FileLocationRequestArgs, DocCommandTemplateResponse } from '../protocol';
+
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+const tryCompleteJsDocCommand = '_typeScript.tryCompleteJsDoc';
+
+
+class JsDocCompletionItem extends CompletionItem {
+	constructor(file: Uri, position: Position) {
+		super('/** @param */', CompletionItemKind.Snippet);
+		this.detail = localize('typescript.jsDocCompletionItem.detail', 'Complete JSDoc comment');
+		this.insertText = '';
+		this.command = {
+			title: 'Try Complete Js Doc',
+			command: tryCompleteJsDocCommand,
+			arguments: [file, position]
+		};
```

Comment:
You are on very thin ice with this... It is really not recommended to modify the document using  [`CompletionItem#command`](https://github.com/Microsoft/vscode/blob/master/src/vs/vscode.d.ts#L2510). Cursor state and undo stack might be confused by this. 

@mjbvz Any reason why you don't use [`CompletionItem#insertText`](https://github.com/Microsoft/vscode/blob/master/src/vs/vscode.d.ts#L2466) and [`CompletionItem#range`](https://github.com/Microsoft/vscode/blob/master/src/vs/vscode.d.ts#L2477). You could create a range that encloses the current, partial doc comment, and replace that with the new text